assets/model_monitoring/components/src/model_data_collector_preprocessor/spark_run.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Entry script for Model Data Collector Data Window Component."""
+
+import argparse
+
+from pyspark.sql import DataFrame
+from pyspark.sql.functions import posexplode, concat_ws, udf, from_json, when, col
+from pyspark.sql.types import StringType
+from dateutil import parser
+from datetime import datetime
+from shared_utilities.momo_exceptions import DataNotFoundError
+from shared_utilities.io_utils import init_spark, save_spark_df_as_mltable
+from shared_utilities.event_utils import add_tags_to_root_run
+from shared_utilities.constants import (
+    MDC_CORRELATION_ID_COLUMN, MDC_DATA_COLUMN, MDC_DATAREF_COLUMN, AML_MOMO_ERROR_TAG
+)
+from mdc_preprocessor_helper import (
+    get_hdfs_path_and_container_client, get_file_list, set_data_access_config
+)
+# used in UDF from executor, need to referenced from root of code
+from model_data_collector_preprocessor.mdc_preprocessor_helper import read_json_content
+
+
+def _mdc_uri_folder_to_raw_spark_df(start_datetime: datetime, end_datetime: datetime, input_data: str,
+                                    add_tags_func=None) -> DataFrame:
+    def handle_data_not_found():
+        add_tags_func({AML_MOMO_ERROR_TAG: "No data found for the given time window."})
+        raise DataNotFoundError(f"No data found for the given time window: {start_datetime} to {end_datetime}")
+
+    add_tags_func = add_tags_func or add_tags_to_root_run
+
+    spark = init_spark()
+    hdfs_path, container_client = get_hdfs_path_and_container_client(input_data, spark=spark)
+    file_list = get_file_list(start_datetime, end_datetime, input_data, hdfs_path, container_client)
+    if not file_list:
+        handle_data_not_found()
+    # print("DEBUG file_list:", file_list)
+
+    set_data_access_config(container_client, spark)
+    df = spark.read.json(file_list)
```

Comment:
Any limits for the file_list?

Reply:
no from spark, it's only limited by memroy, and the OS limit for concurrent opened files.
and I have a todo to read per day instead of per hour and then filtered by time